src/mpid/ch4/src/ch4_impl.h

```diff
@@ -963,27 +963,25 @@ static inline int MPIDI_CH4U_allocate_shm_segment(MPIR_Comm * shm_comm_ptr,
 {
     MPIR_Errflag_t errflag = MPIR_ERR_NONE;
     int mpi_errno = MPI_SUCCESS;
-    MPL_shm_hnd_t shm_segment_handle;
-    void *map_ptr = NULL;
```

Comment:
I assume Coverity was complaining about this `map_ptr`. I don't understand why this patch fixes a resource leak issue. Can you explain more?

Reply:
I improved the commit message. Could you please check if it makes sense to you ? Besides, I found the caller functions have similar issue (i.e., used a temporary variable and jumped to `fn_fail` before set the value to window, thus segment info lost). I just fixed it.

Reply:
Oh I see, so the actual error handing (including freeing already allocated shm segments) are done later at `MPIDI_CH4R_win_finalize`.  I was missing that part. Now it makes sense.